controllers/k8ssandra/datacenters.go

```diff
@@ -93,6 +103,18 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 
 		if err = remoteClient.Get(ctx, dcKey, actualDc); err == nil {
 			// cassdc already exists, we'll update it
+
+			if _, rebuildNeeded = actualDc.Labels[api.RebuildLabel]; rebuildNeeded {
```

Comment:
Just above (L85) the rebuildNeeded is parsed. Here we overwrite whatever was parsed there and again set the desiredDc.Labels one. Why?

Reply:
I initially determine if a rebuild is needed by looking at the K8ssandraCluster status. If the cluster is already initialized and if dc is not present in the status, then the rebuild label is added. On subsequent reconciliations the datacenter will be added to the status, well before the rebuild happens. The initial evaluation will then result in `rebuildNeeded == false`. This is why it is overwritten based on the presence of the label. After the rebuild finishes, the label is removed.

Do you want me to provide some additional comments explaining this?

Reply:
The logic is quite cumbersome. An annotation on the `kc` would probably simplify it a good bit. I'll look into making the change.